contrib/ceph-versioned-image-descriptions.md

```diff
@@ -0,0 +1,55 @@
+DockerHub descriptions for ceph-vX.Y.Z images
+==============================================
+
+ceph/ceph
+----------
+
+### Short Description
+Images containing all Ceph binaries.
+
+### Full Description
+Images contain all Ceph binaries as well as NFS-Ganesha and iSCSI binaries since these components
+are heavily tied to Ceph's version to be compatible. Unless otherwise noted in the tag, all images
+are CentOS based. New images are built within 24 hours of a new Ceph version being published to
+the official Ceph package repository.
+
+#### Image tag breakdown
+
+**Build date** </br>
+Some images are suffixed with an 8-digit build date in the form `YYYMMDD`. This indicates the date
+the image was built and is used similarly to a build number for typical packages. For an image which
+already exists (e.g., `v12.2.7-20181023`), a new version with a different build date will be built
+when there is a base image update. For example, if the CentOS base image gets a security fix on
+10 February 2080, the example image above will get a new image built with tag `v12.2.7-20800210`.
+
+**Versions** </br>
+There are a few ways to choose the Ceph version you desire:
+ - Full semantic version with build date, e.g., `v12.2.9-20181026`
+   - These tags are intended for use when precise control over image upgrade is desired and are
+     recommended for production use.
+ - Major version, e.g. `v12` (a.k.a. _Ceph Luminous_)
+   - These tags are always the most recent build of the newest Ceph major release matching the tag.
+ - Minor version (e.g., `v12.1`)
+   - These tags are always the most recent build of the newest Ceph minor release matching the tag
+     for environments where more precise control is needed than a major version but where bug fixes
+     both in Ceph and in the base image are desired.
+
```

Comment:
what about the tag `v12.2.9`, which represents the latest build for that patch release? So if we build that patch again in the future with a different base os update, this tag can still be used. Not high priority, but seems like a nice-to-have.

Reply:
Tags that are just `v12.2.9` don't exist, my assumption being if a user wants to be that specific about the Ceph version, they likely want to be that specific about the build version. This can be changed, but that involves code changes.

Reply:
Those are the public versions that everyone uses.  I'm baffled as to why they're not the common case.

Reply:
This, again, can be changed, but is out of scope here, and this seems to be delaying us from getting documentation of any kind on these images on Docker Hub.

Reply:
I don't see why it's delaying putting the doc up, but, I was asked for comments, and I didn't believe those comments needed to be limited to just the documentation.  Issues are issues, regardless of the particular file they arise from.

Since the subject has come up, however, I believe the current tagging system is so deficient and confusingly done that I believe it would be better to fix the tagging system than to document it, as broken as it is.  Others may, of course, override my opinion, and I'm not going to fight to the death about it.